packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -262,28 +262,60 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
      * This is the generic `Open` method. Opens files and directories too. Resolves to the opened URI.
      * Except when you are on either Windows or Linux `AND` running in electron. If so, it opens a file.
      */
-    protected async doOpen(): Promise<URI | undefined> {
+    protected async doOpen(): Promise<MaybeArray<URI> | undefined> {
         if (!isOSX && this.isElectron()) {
             return this.doOpenFile();
         }
         const [rootStat] = await this.workspaceService.roots;
         const destinationUri = await this.fileDialogService.showOpenDialog({
             title: WorkspaceCommands.OPEN.dialogLabel,
             canSelectFolders: true,
-            canSelectFiles: true
+            canSelectFiles: true,
+            canSelectMany: true
         }, rootStat);
-        if (destinationUri && this.getCurrentWorkspaceUri()?.toString() !== destinationUri.toString()) {
-            const destination = await this.fileService.resolve(destinationUri);
-            if (destination.isDirectory) {
-                this.workspaceService.open(destinationUri);
+        if (destinationUri) {
+            if (Array.isArray(destinationUri)) {
+                // only open files then open all folders in a new workspace, as done with Electron see doOpenFolder
+                const folders: MaybeArray<URI> = [];
+                for (const uri of destinationUri) {
+                    if (this.getCurrentWorkspaceUri()?.toString() !== uri.toString()) {
```

Comment:
This check is only necessary in the 'folder' case. If one of the selected file is a workspace file, we still want to open it as a file, not a workspace. Performance is not an issue here, since we're doing this a few times max per user interaction.

Reply:
Check has been moved to the folder case only. It is now possible to open a workspace file as a simple file.